metadata_writer_nne.c

```diff
@@ -94,132 +309,725 @@ static uint8_t md_nne_handle_gps_event(struct md_writer_nne *mwn,
     return RETVAL_SUCCESS;
 }
 
-static void md_nne_handle_timeout(void *ptr) {
-    struct md_writer_nne *mwn = ptr;
+/*static void md_nne_handle_conn_event(struct md_writer_nne *mwn,
+                                     struct md_conn_event* mce) {
+
+    const char* const conn_event_type[] = {
+        "",
+        "CONN_EVENT_L3_UP",
+        "CONN_EVENT_L3_DOWN",
+        "CONN_EVENT_L4_UP",
+        "CONN_EVENT_L4_DOWN",
+        "CONN_EVENT_MODE_CHANGE",
+        "CONN_EVENT_QUALITY_CHANGE",
+        "CONN_EVENT_META_UPDATE",
+        "CONN_EVENT_META_MODE_UPDATE",
+        "CONN_EVENT_META_QUALITY_UPDATE",
+    };
+
+    META_PRINT_SYSLOG(mwn->parent, LOG_ERR, "NNE writer: md_conn_event %s: interface_name=%s, imei=%s, network_address=%s, event_value=%d, event_value_str=%s",
+                      conn_event_type[mce->event_param],  mce->interface_name, mce->imei, mce->network_address, mce->event_value, mce->event_value_str);
+}*/
+
+static struct nne_modem *md_nne_get_modem(struct nne_modem_list *modem_list, uint32_t mccmnc)
+{
+    struct nne_modem *e = NULL;
+    for (e = modem_list->lh_first; e != NULL; e = e->entries.le_next)
+        if (e->mccmnc == mccmnc)
```

Comment:
Ok, thanks for letting me know. I just thought that maybe we could make a better solution (that will be backwards compatible), so we dont have to go back and change this later. But this is not critical.